Frameworks/Foundation/NSDecimalNumber.mm

```diff
@@ -14,296 +14,344 @@
+    return s_maxNumber;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalMultiplyByPowerOf10(NSDecimal* result, const NSDecimal* number, short power, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)minimumDecimalNumber {
+    static NSDecimalNumber* s_minNumber =
+        [[NSDecimalNumber alloc] initWithMantissa:std::numeric_limits<unsigned long long>::max() exponent:_MAX_EXP isNegative:YES];
+    return s_minNumber;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalNormalize(NSDecimal* number1, NSDecimal* number2, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)decimalNumberWithDecimal:(NSDecimal)decimal {
+    return [[[NSDecimalNumber alloc] initWithDecimal:decimal] autorelease];
+}
+
+/**
+ @Status Interoperable
+*/
++ (NSDecimalNumber*)decimalNumberWithMantissa:(unsigned long long)mantissa exponent:(short)exponent isNegative:(BOOL)isNegative {
+    return [[[NSDecimalNumber alloc] initWithMantissa:mantissa exponent:exponent isNegative:isNegative] autorelease];
 }
 
 /**
  @Status Stub
  @Notes
 */
-NSCalculationError NSDecimalPower(NSDecimal* result, const NSDecimal* number, NSUInteger power, NSRoundingMode roundingMode) {
++ (NSDecimalNumber*)decimalNumberWithString:(NSString*)numericString locale:(id)locale {
     UNIMPLEMENTED();
     return StubReturn();
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-void NSDecimalRound(NSDecimal* result, const NSDecimal* number, NSInteger scale, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
+- (NSNumber*)initWithBool:(BOOL)value {
+    return [self initWithLongLong:static_cast<long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSString* NSDecimalString(const NSDecimal* dcm, id locale) {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithChar:(char)value {
+    return [self initWithLongLong:static_cast<long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalSubtract(NSDecimal* result,
-                                     const NSDecimal* leftOperand,
-                                     const NSDecimal* rightOperand,
-                                     NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithDouble:(double)value {
+    int exp = 0;
+    double res = value;
+    if (res < 0) {
+        res *= -1;
+    }
+
+    if (res == DBL_MIN) {
+        return [self initWithMantissa:0 exponent:0 isNegative:0];
+    }
+
+    if (res > _DOUBLE_LIMIT) {
+        return [self initWithMantissa:ULLONG_MAX exponent:0x7f isNegative:(value < 0)];
+    }
+    // Attempt best for smaller numbers (this is being compacted via NSDecimalCompact in initWithMantissa)
+    while ((res != 0) && (res < static_cast<double>(ULLONG_MAX))) {
```

Comment:
okay I remember it now (it's been awhile). this was to make sure that the floating point precision of double was saved. pretty much shift the value.
Now this was to match the reference platform specifically, 

---
In reply to: [81253840](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r81253840) [](ancestors = 81253840,81253259,81252442,81252229,81252181,81251181,81246684)